### ISSUE TITLE
ALE does not call for golangci-lint

### ISSUE THREAD

**Issue by konart** [dense-analysis/ale#3330]
## Information

**VIM version**

NVIM v0.4.4
Build type: Release

Operating System: macOS 10.15.3

## What went wrong

No ALE output in any form despite golangci-lint output to stdout

## Reproducing the bug

1. Open *.go file in an existing project
2. Make some changes that should trigger linter and return some issues

### :ALEInfo

 Current Filetype: go
Available Linters: ['bingo', 'gobuild', 'gofmt', 'golangci-lint', 'golint', 'gometalinter', 'gopls', 'gosimple', 'gotype', 'govet', 'golangserver', 'revive', 'staticcheck']
   Linter Aliases:
'gobuild' -> ['go build']
'govet' -> ['go vet']
  Enabled Linters: ['gofmt', 'golangci-lint']
 Suggested Fixers: 
  'gofmt' - Fix Go files with go fmt.
  'goimports' - Fix Go files imports with goimports.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_go_go_executable = 'go'
let g:ale_go_golangci_lint_executable = 'golangci-lint'
let g:ale_go_golangci_lint_options = ''
let g:ale_go_golangci_lint_package = 0
let g:ale_go_langserver_executable = 'go-langserver'
let g:ale_go_langserver_options = ''
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 0
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 0
let g:ale_linter_aliases = {}
let g:ale_linters = {'go': ['golangci-lint', 'gofmt']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 1
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 0
let g:ale_set_quickfix = 1
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) gofmt
(finished - exit code 0) ['/usr/local/bin/zsh', '-c', '''gofmt'' -e ''/var/folders/6y/7v29j_w140jb9f_9tn90yrlm0000gp/T/nvimVTopWh/3/api.go''']

<<<NO OUTPUT RETURNED>>>

As you can see there is no `(executable check - success)` info about golangci-lint, however there is no problem running it as is. 

ALEInfo looks better if I remove those options from my .vimrc:
```
" let g:ale_lint_on_enter = 0
" let g:ale_lint_on_save = 1
" let g:ale_lint_on_text_changed = 0
" let g:ale_set_quickfix = 1
" let g:ale_open_list = 1
```

  Command History:
(executable check - success) gofmt
(finished - exit code 0) ['/usr/local/bin/zsh', '-c', '''gofmt'' -e ''/var/folders/6y/7v29j_w140jb9f_9tn90yrlm0000gp/T/nvimrB669A/3/api.go''']
<<<NO OUTPUT RETURNED>>>
(executable check - success) golangci-lint
(finished - exit code 1) ['/usr/local/bin/zsh', '-c', 'cd ''/Users/konart/build/goods/neomis/internal/httpd/api/v1'' && ''golangci-lint'' run ''api.go'' ']
<<<OUTPUT STARTS>>>
api.go:23: File is not `gofumpt`-ed (gofumpt)
<<<OUTPUT ENDS>>>

**Comment by konart** [comment]
After some testing it seems like disabling `ale_lint_on_enter` disables golangci-lint somehow.

**Comment by gerardabello** [comment]
I'm experiencing the same problem. I see nothing in the `:ALEInfo`, even with history enabled. In addition, when trying to run the linter manually by `:ALELint golangci_lint`, I get: `E488: Trailing characters`.

**Comment by hsanson** [comment]
This linter has `lint_file` flag set to 1 so linting will only trigger on events that occur against the file, not the buffer.

See `:h ale#linter#Define()`, specifically the lint_file flag explanation:

```
Linters where the eventual value of this option
evaluates to 1 will not be run as a user types, per
g:ale_lint_on_text_changed. Linters will instead be
run only when events occur against the file on disk,
including g:ale_lint_on_enter and
g:ale_lint_on_save. Linters where this option
evaluates to 1 will also be run when the ALELint
command is run.
```
